src/components/Menu/Menu.minors.tsx

```diff
@@ -33,11 +33,21 @@ export function Item({
   href,
   toggle = false,
   icon,
+  /** Function called if item is toggled open */
+  onOpen,
+  /** Function called if item is toggled close */
+  onClose,
   ...props
 }) {
   const [open, setOpen] = useState(!toggle);
 
-  const doToggle = () => toggle && setOpen((open) => !open);
+  const doToggle = () =>
+    toggle &&
+    setOpen((open) => {
+      // Since state hasn't changed yet, open will still be false here
+      open ? onClose && onClose() : onOpen && onOpen();
```

Comment:
Can we call `onOpen` & `onClose` in a `useEffect` here? State setters are meant to be pure functions and should definitely not have side effects.

Reply:
Just updated! If there's a better way to prevent calling `onClose` on mount... let me know. 🤔